### PR TITLE
bodyswap: handle companions

### PR DESCRIPTION
**Pull request by AtomicChicken**
Requires DFHack/df-structures#426.

Fix: 
- Prior party members no longer tag along after bodyswapping and reloading the map.

Improvement:
- Companions of bodyswapping targets are now added to the adventurer party (when swapping into a member of a previous party, for example) and can be swapped into using the in-game tactical party system.

### CODE REVIEW

File: bodyswap.lua [hunk 1]
```diff
@@ -75,13 +75,64 @@ function createNemesis(unit)
...
+end
+
+function processNemesisParty(nemesis, targetUnitID, alreadyProcessed)
+-- configures the target and any leaders/companions to behave as cohesive adventure mode party members
+  local alreadyProcessed = alreadyProcessed or {}
+  alreadyProcessed[tostring(nemesis.id)] = true
+
+  local nemUnit = nemesis.unit
+  if nemesis.unit_id == targetUnitID then -- the target you're bodyswapping into
+    df.global.ui_advmode.interactions.party_core_members:insert('#', nemesis.figure.id)
```
Comment (lethosor) [inline bodyswap.lua:100]: I don't have an adventure save handy to test this: do you have any idea if DF expects these vectors of IDs to be sorted? If so, `utils.insert_sorted()` might be a good idea in these cases.

Comment (AtomicChicken) [reply bodyswap.lua:100]: My impression is that it doesn't based on the observation that the IDs in `party_extra_members` and `party_pets` are ordered according to the sequence in which the corresponding figures joined the party, though it's difficult to say for certain with `party_core_members` as additional figures aren't added to this list during normal gameplay.

Comment (lethosor) [reply bodyswap.lua:100]: Ok! Hopefully DF is consistent, then.

**Review by lethosor — APPROVED** [review]
Managed to swap between party members a couple times - looks good!

**Pull request merged**